File: connect4/evaluators/empty.py
"""
Starter evaluator for student heuristic design.

Add reusable heuristic helper functions in connect4/heuristics.py,
then combine them here to build a stronger evaluator.
"""

from ..board import GameState
from ..heuristics import terminal_state_bonus


class RandomEvaluator:
    """Baseline evaluator built from heuristic helper functions."""

    def evaluate(self, state: GameState, player: int) -> float:
        score = 0.0
        score += terminal_state_bonus(state, player)
        return score

    def __repr__(self) -> str:
        return "Empty"


EVALUATOR = RandomEvaluator()

File: evaluators/empty.c
/*
 * Starter evaluator — baseline that only checks terminal states.
 *
 * Copy this file to create a new evaluator, then add your heuristics.
 */

#include "../heuristics.h"

float empty_evaluate(const GameState *state, int player) {
    float score = 0.0f;

    score += terminal_state_bonus(state, player);

    /* Add your heuristics here, e.g.:
     * score += 2.0f * center_control(state, player);
     * score += 5.0f * threat_detection(state, player);
     */

    return score;
}
